core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/upscale/StackUpscaleActions.java

```diff
@@ -120,7 +120,7 @@ protected void doExecute(StackScalingFlowContext context, StackScaleTriggerEvent
                         payload.getAdjustment(), context.isRepair());
                 stackUpscaleService.addInstanceFireEventAndLog(context.getStack(), payload.getAdjustment(), payload.getInstanceGroup(),
                         payload.getAdjustmentTypeWithThreshold());
-                if (instanceCountToCreate > 0) {
+                if (context.isRepair() || instanceCountToCreate > 0) {
```

Comment:
Please rabase to the latest master, because the context.isRepair is already used during instanceCountToCreate calculation.